src/storage/src/source/postgres/snapshot.rs

```diff
@@ -321,13 +321,39 @@ pub(crate) fn render<G: Scope<Timestamp = MzOffset>>(
             }
             *rewind_cap_set = CapabilitySet::new();
 
-            let upstream_info = mz_postgres_util::publication_info(
+            let upstream_info = match mz_postgres_util::publication_info(
                 &context.ssh_tunnel_manager,
                 &connection_config,
                 &connection.publication,
                 None,
             )
-            .await?;
+            .await
+            {
+                // If the replication stream cannot be obtained in a definite way there is
+                // nothing else to do. These errors are not retractable.
+                Err(err)
```

Comment:
One last wish to define a tiny error type

```rust
#[this_error::Error]
enum PublicationInfoError {
    PublicationMissing,
    PostgresError(#[from] PostgresError),
}
```

 to avoid string matching on error strings which could change in the future. What do you think?

Reply:
Pushed a commit that does this. I avoided it originally because of how it infects so many call sites with the need to provide a conversion for new error type (as opposed to the likelihood that these strings drift). lmk if you envisioned something different that I missed.

Reply:
I hadn't foreseen this, I guess you could also have flattened the two enums but mapping the error is totally fine.